### PR TITLE
Chapter_9

### PR DESCRIPTION
**Pull request by letterforyou26**
Chapter 9: Advanced login
Link câu hỏi mỗi chapter: 
https://docs.google.com/document/d/1CrWvmYMJpn0kiq-oyPuEkD8R8-K_QUL7hfgJtILPthE/edit?fbclid=IwAR2uaR7FELR0-XJZD2t2QdMUutM_QsX5BDfm94qbo6I9dCarjT3xmYbpqXM#

### CODE REVIEW

**Comment by letterforyou26** [comment]
success

**Comment by letterforyou26** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -22,8 +24,37 @@ class User < ApplicationRecord
...
   has_secure_password
 
-  private
+  class << self
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+               BCrypt::Engine::MIN_COST
+             else
+               BCrypt::Engine.cost
+             end
```
Comment (ghost) [inline app/models/user.rb]:
```suggestion
      min_cost = ActiveModel::SecurePassword.min_cost
      cost = min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
```

**Review by ghost — COMMENTED** [review]
cmt

**Comment by letterforyou26** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**